transport-classes-io_uring/src/main/java/io/netty/incubator/channel/uring/IOUringEventLoop.java

```diff
@@ -166,7 +166,10 @@ protected void run() {
                 if (curDeadlineNanos == -1L) {
                     curDeadlineNanos = NONE; // nothing on the calendar
                 }
-                nextWakeupNanos.set(curDeadlineNanos);
+
+                // The memory ordering effects of the nextWakeupNanos are irrelevant. A setOpaque should be sufficient,
+                // but on Java 8 the lazySet is the cheapest alternative. It is free on X86 since every store is a release store.
+                nextWakeupNanos.lazySet(curDeadlineNanos);
```

Comment:
@pveentjer we must ensure that this write happens before the subsequent `hasTasks()` read or else a (non-IO) task could be put in the queue by another thread without us being awoken.

Note that the cost here should not be significant anyhow. We are about to make a blocking syscall because there's no more work to do, so it is not on a critical path. A heavily loaded system should rarely reach here since we'll remain in the CQ processing loop below.

Reply:
So you need to order an older store with a newer load to a different address. So you need a [StoreLoad] which is provided by a volatile write and not by something weaker. So my fix would break the code.

Reply:
@pveentjer so I guess we should close this pr ?